### PR TITLE
Hook router into state + open Belt w/ compiler flag

### PR DESCRIPTION
**Pull request by IwanKaramazow**
- `open Belt` by default with compiler flag
- simplify a pattern match
- Connect the app navigation to the url, through Reason's router:

1) ReasonReact.Router.push(url) triggers all subscriptions
2)
```reason
      ReasonReact.Router.watchUrl(url =>
        self.send(ChangeRoute(T.Route.urlToRoute(url)))
      );
```
3)  the reducer in `app.re` receives the `ChangeRoute` action, and we update the state

### CODE REVIEW

**Comment by kutyel** [comment]
Awesome, thank you very much! 👏🎉

File: src/post.re [hunk 1]
```diff
@@ -16,7 +16,7 @@ let make = (~post, ~onLike, ~onClick, _children) => {
     let likes = string_of_int(count);
     <figure className="grid-figure">
       <div className="grid-photo-wrap">
-        <a href={j|/view/$id|j} onClick={onClick(Detail(id))}>
+        <a href={j|/view/$id|j} onClick={navigate(Route.Detail(id))}>
```
Comment (kutyel) [inline src/post.re:19]: How is it that here the `Route` type is needed, and in L34 it is not? 🤔

Comment (IwanKaramazow) [reply src/post.re:19]: OCaml tries to use locally available information to disambiguate between the various fields and constructors. The compiler makes a decision at L19 based on the `Route` module identifier.
The type is fixed here. From that point on in `post.re`, `Detail` is going to refer to `Route.Detail`.

**Review by kutyel — APPROVED** [review]
🎉 👍

**Pull request merged**